### PR TITLE
Elyra Canvas Documentation using mkdocs

### PR DESCRIPTION
**Pull request by srikant-ch5**
Fixes: #1701
 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Comment by srikant-ch5** [comment]
Hi @matthoward366 Please review above setup for new Elyra Canvas Documentation

**Comment by srikant-ch5** [comment]
> @srikant-ch5 Is there a way I can test this out locally?

Hi @matthoward366 Yes you can run this using command mkdocs serve .

**Comment by tomlyn** [comment]
@matthoward366 
For getting the docs build to work @Paul-Kawalkowski helped me discover that:

1.  Python v3 needs to be available. I had v3.11.3 installed
2. Then run  `pip3 install -r requirements.txt`
3. Then run `mkdocs serve`
4. When complete, open the browser:  http://127.0.0.1:8000/

@srikant-ch5 Can you add this info to the Read me?

**Comment by srikant-ch5** [comment]
> @matthoward366 For getting the docs build to work @Paul-Kawalkowski helped me discover that:
> 
> 1. Python v3 needs to be available. I had v3.11.3 installed
> 2. Then run  `pip3 install -r requirements.txt`
> 3. Then run `mkdocs serve`
> 4. When complete, open the browser:  http://127.0.0.1:8000/
> 
> @srikant-ch5 Can you add this info to the Read me?

Hi @tomlyn . Sure, I have added above instructions in Readme.

File: mkdocs.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+extra_css:
+  - css/styles.css
+
+extra_javascript:
+  - js/links-in-new-tabs.js
+
+edit_uri: edit/main/guide/
+
+# Documentation and theme
+copyright: 'Copyright &copy; 2023 IBM Watson Team, an open source project and all are welcome to contribute!<br> Maintained by the SPSS Modeler Team'
```
Comment (nmgokhale) [inline mkdocs.yml:27]: Update year to 2024. Maintained by the elyra canvas Team.

File: mkdocs.yml [hunk 2]
```diff
@@ -0,0 +1,80 @@
+site_name: Elyra Guide
```
Comment (nmgokhale) [inline mkdocs.yml:1]: This should be `Elyra Canvas Guide`

File: README.md [hunk 3]
```diff
@@ -50,3 +50,15 @@ Delete @elyra/canvas from node_modules of your application
 # Run npm install to get local copy of common-canvas and/or common-properties
 npm install
 ```
+
+### Contribute to the Elyra Canvas documentation (mkdocs)
+1. Python v3 needs to be available.
```
Comment (nmgokhale) [inline README.md:55]: Add a step - 
2. Go to `canvas` directory
3. Install required mkdocs packages using pip3...

File: guide/5.0-Styling.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+$font-path: "/fonts";
+@import "carbon-components/scss/globals/scss/styles";
+```
+The fonts will need to be imported from carbon before carbon styles and placed in a public `/fonts` directory.
+You can see an example of this in the common-canvas test harness (which is the equivalent of a host application) in this repo. That is, the [index.scss file](https://github.com/elyra-ai/canvas/blob/master/canvas_modules/harness/src/styles/index.scss) contains the lines above and the grunt build files ensures the fonts are copied from`node_modules/carbon-components/src/globals/` to the `<carbon fonts folder>` directory.
+
+
+### 3rd party styling
+
+If you are using the expression builder in common-properties then also include these libraries:
```
Comment (nmgokhale) [inline guide/5.0-Styling.md:44]: Remove lines 44-47 -

```
If you are using the expression builder in common-properties then also include these libraries:

codemirror/addon/hint/show-hint.css
codemirror/lib/codemirror.css
```

**Comment by srikant-ch5** [comment]
Hi @nmgokhale Thanks for the review. I have updated the PR with above feedback.

**Review by tomlyn — DISMISSED** [review]

**Comment by srikant-ch5** [comment]
Hi Team, I have deployed the docs at https://srikant-ch5.github.io/canvas/ Please review. The url can be changed to elyra-canvas .

**Review by tomlyn — DISMISSED** [review]

File: docs/3.3-Common-Properties-Controls.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+* `actionPanel` A panel used for containing action controls.
+* `textPanel` A panel used to display static label and/or description. 
+* `twistyPanel` A panel used to display a panel title that expands to the panel content when clicked.
+* `columnPanel` A panel used to display children side by side. 
+* `tearsheetPanel` A tearsheet panel.  The panel can be opened/closed using the propertyController methods `setActiveTearsheet(groupId)` and `clearActiveTearsheet()`
+
+## Parameter Controls
+The following controls are supported in the Common Properties editor. Control types are intended for use with particular parameter types:
+
+### Control Types
```
Comment (nmgokhale) [inline docs/3.3-Common-Properties-Controls.md:27]: `slider` control documentation is not added in this section.

Comment (srikant-ch5) [reply docs/3.3-Common-Properties-Controls.md:27]: > `slider` control documentation is not added in this section.

Hi @nmgokhale I have included slider control documentation now https://srikant-ch5.github.io/canvas/3.3-Common-Properties-Controls/#control-types . Previously I was taking https://github.com/elyra-ai/canvas/wiki as reference (slider is not included in this).

**Review by nmgokhale — APPROVED** [review]

File: canvas_modules/common-canvas/src/object-model/layout-dimensions.js [hunk 6]
```diff
@@ -73,7 +73,7 @@ const portsHorizontalDefaultLayout = {
 		labelAllowReturnKey: false, // true allows line feed to be inserted into label, "save" to make the return key save the label.
 
 		// An array of decorations to be applied to the node. For details see:
-		// https://github.com/elyra-ai/canvas/wiki/2.4.2-Decoration-Specification
+		// https://github.com/elyra-ai/canvas/tree/main/guide/2.4.2-Decoration-Specification.md
```
Comment (matthoward366) [inline canvas_modules/common-canvas/src/object-model/layout-dimensions.js:76]: Shouldn't all these links be github pages links?  Also, I don't this `guide` is the folder anymore.

File: canvas_modules/common-canvas/src/object-model/layout-dimensions.js [hunk 7]
```diff
@@ -488,7 +488,7 @@ const portsVerticalDefaultLayout = {
...
```
Comment (matthoward366) [inline canvas_modules/common-canvas/src/object-model/layout-dimensions.js:491]: Same as above.  Have a look at all the links.

File: docs/css/styles.css [hunk 8]
```diff
@@ -0,0 +1,7 @@
+th {
```
Comment (matthoward366) [inline docs/css/styles.css:1]: Shouldn't all these new css and js files have copyrights?

File: docs/images/DELETEME [hunk 9]
```diff
@@ -0,0 +1 @@
+Images for the guide go here.
```
Comment (matthoward366) [inline docs/images/DELETEME:1]: What is this?

File: docs/js/links-in-new-tabs.js [hunk 10]
```diff
@@ -0,0 +1,10 @@
+// This takes any external links and adds _blank to make sure they open in a new tab. Links to other
+// pages within the documentation will remain in the same tab
+// Source: https://stackoverflow.com/questions/4425198/can-i-create-links-with-target-blank-in-markdown#answer-4425214
+
+var links = document.links;
```
Comment (matthoward366) [inline docs/js/links-in-new-tabs.js:5]: Missing copyright.  Also, we usually use `const` instead of `var`

**Review by matthoward366 — CHANGES_REQUESTED** [review]

File: docs/images/DELETEME (hunk 9 above)
Comment (srikant-ch5) [reply docs/images/DELETEME]: Hi Matt I have removed this file now. Thanks.

File: canvas_modules/common-canvas/src/object-model/layout-dimensions.js (hunk 7 above)
Comment (srikant-ch5) [reply canvas_modules/common-canvas/src/object-model/layout-dimensions.js]: Hi Matt, I have reverted the docs links to point to existing elyra canvas docs once the new one is deployed I will change these urls pointing to new docs.

File: canvas_modules/common-canvas/src/object-model/layout-dimensions.js (hunk 6 above)
Comment (srikant-ch5) [reply canvas_modules/common-canvas/src/object-model/layout-dimensions.js]: Hi Matt, I have reverted the docs links to point to existing elyra canvas docs once the new one is deployed I will change these urls pointing to new docs.

File: .github/workflows/ci.yml [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+permissions:
+  contents: write
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Configure Git Credentials
+        run: |
+          git config user.name github-actions[bot]
```
[the remark this answers is not in the document]
Comment (srikant-ch5) [reply .github/workflows/ci.yml:15]: Hi Matt to deploy the docs using gh-pages I have followed https://squidfunk.github.io/mkdocs-material/publishing-your-site/ resource and able to deploy whenever I push any change in docs to my new-guide branch. It triggers ci to deploy the changes without manually running `mkdocs gh-deploy --force`
Screenshot 2024-02-29 at 7 12 06 PM

File: .github/workflows/ci.yml [hunk 12]
```diff
@@ -0,0 +1,28 @@
+name: ci 
+on:
+  push:
+    branches:
+      - new-guide
```
[the remark this answers is not in the document]
Comment (srikant-ch5) [reply .github/workflows/ci.yml:5]: Yes Matt, this needs to be changed as main when merged, for testing purpose I have kept it as new-guide.

File: docs/js/links-in-new-tabs.js (hunk 10 above)
Comment (srikant-ch5) [reply docs/js/links-in-new-tabs.js]: Thanks, I have updated these.

File: .github/workflows/ci.yml (hunk 11 above)
[the remark this answers is not in the document]
Comment (matthoward366) [reply .github/workflows/ci.yml:15]: I'm more wondering about `git config user.name github-actions[bot]` and `git config user.email 41898282+github-actions[bot]@users.noreply.github.com`.  Do we need these?

**Review by matthoward366 — COMMENTED** [review]
I small comment and the rest looks good.  Once it's deployed with your branch then we can change it to main before merging.

**Comment by srikant-ch5** [comment]
> I'm more wondering about git config user.name github-actions[bot] and git config user.email 41898282+github-actions[bot]@users.noreply.github.com. Do we need these?

Hi @matthoward366 I have removed these and its working fine. Thanks.

**Review by matthoward366 — APPROVED** [review]

**Pull request merged**